cs244b_project/src/lib.rs

```diff
@@ -6,8 +6,10 @@ mod utils;
 
 use itertools::Itertools;
 use rand::Rng;
+use tokio::io::AsyncWriteExt;
```

Comment:
FYI, this is not a cancellation-safe method, so we can't call it as an option in select!. (Which you're not, just noting in case we refactor.)